CryptoLightNews/Networking/Service.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Service.swift
+//  CryptoLightNews
+//
+//  Created by Richard Price on 21/12/2022.
+//
+
+import Foundation
+
+class Serivce {
+    static let shared = Serivce()
+    typealias News = Result<SearchResults, CryptoNewsErrors>
+    let constants = Constants()
+    
+    func fetchNews(completion: @escaping (News) -> Void) {
+        let urlString = CryptoNewsAPI.getNews
+        fetchGenericJSONData(urlString: urlString.rawValue, completion: completion)
+    }
+    
+    func fetchGenericJSONData<T: Decodable>(urlString: String, completion: @escaping (Result<T, CryptoNewsErrors>) -> ()) {
+        guard let url = URL(string: urlString) else {
+            completion(.failure(CryptoNewsErrors.networkError))
+            return
+        }
```

Comment:
Put it in the `fetchNews`
```suggestion
       
```

Reply:
struggling with this one unable to get it to work with the suggestion